server/src/main.rs

```diff
@@ -243,6 +243,23 @@ fn main() {
       // RSS
       .route("/feeds/{type}/{name}.xml", web::get().to(feeds::get_feed))
       .route("/feeds/all.xml", web::get().to(feeds::get_all_feed))
+      // Federation
+      .route(
+        "/.well-known/webfinger",
+        web::get().to(apub::webfinger::get_webfinger),
+      )
+      .route(
+        "/api/v1/federation/community/{community_name}",
+        web::get().to(apub::community::get_apub_community)
+      )
```

Comment:
This is not how Mastodon does it, so I believe it would not be compatible. Mastodon serves this data with a .json extension, however the URLs listed in the Actor definition do not have the .json extension. So you should add the .json part only here in the routing.

Also, Mastodon exposes this stuff under `users/name/xxx`, not under the API, so I was thinking to use URLs like `/c/main/xxx` for Lemmy. 

Check out my actor for reference: https://radical.town/users/felix.json